cohortextractor2/query_language.py

```diff
@@ -79,10 +79,31 @@ def __invert__(self):
             rhs=self.rhs,
         )
 
+    def __gt__(self, other):
+        return self._compare("__gt__", other)
+
+    def __ge__(self, other):
+        return self._compare("__ge__", other)
+
+    def __lt__(self, other):
+        return self._compare("__lt__", other)
+
+    def __le__(self, other):
+        return self._compare("__le__", other)
+
+    def __eq__(self, other):
+        return self._compare("__eq__", other)
+
+    def __ne__(self, other):
+        return self._compare("__ne__", other)
```

Comment:
I don't understand what it means to carry about comparison operations on a comparator object. In my head a comparator effectively has a boolean type, so the logical operations implemented previously make sense, but these comparisons are surprising.

Reply:
You're right, it doesn't make sense to compare comparators (although you can compare boolean values with things, not that that makes any more sense really). 

I put this in to appease mypy, because if a `PatientSeries.value` can be either a `Value` or a `Comparator`, it complains if you try to do one of these operations on a `PatientSeries`.  I tried making a `ValueFromComparator` instead, so that a `PatientSeries` just takes a `Value` but then I just ran into the same issue further down the line, and it got more complicated.

An option might be to raise an exception if you try to use any of these methods?  So if you tried to do `(age == 20) < (age == 30)` in a mapping, it'll raise an exception because that's nonsense?

Reply:
I really hope we don't have to add un-callable methods just to appease mypy. If it thinks they need to be there but we never call them then we must be lying to it somewhere. I don't know what else to suggest though, without having seen the details. Maybe we could look at this together tomorrow?

Reply:
I added another commit that raises exceptions, which I think is better than letting them do nonsense things, but not ideal.  We don't call those methods, but we could.  We don't call them only because it doesn't make logical sense to call them, in the same way that it doesn't really make sense to say `(1 == 2) > 3` but python allows it (it's False, apparently).  And it's perfectly possible that a researcher might write something like that in error in a data definition, and it'd make more sense to tell them they couldn't do that than to let it pass without comment.